### ISSUE TITLE
Memory issue with double data type

### ISSUE THREAD

**Issue by mavencode01** [graphframes/graphframes#168]
I've been wondering why am getting memory issue for the past couple of days in my spark job when I've more than enough memory to run entire job - `400G, 120Cores, 1TB disk space.`

It turns out that the Dataset contains a `Double` data type and causes some memory issue when running Connected Component algorithm

    org.apache.spark.shuffle.MetadataFetchFailedException: Missing an output 
    location for shuffle 4
    ....

This particular problem occurs when processing a large dataset - 143M records.

The temporary solution was to ignore and set the values to `0.0f` to fix the problem

    ...
    case class GraphVertex(id: String, path: String, score: Double, relative_score: Double, size: Integer)
    ...

    //genes.map(v => GraphVertex(v.trait, v.loca, v.score, v.relative_score, v.size))
    //      .dropDuplicates                                
    //     .alias("v")

    //temporary fix
    genes.map(v => GraphVertex(v.trait, v.loca, 0.0, 0.0, v.size))
         .dropDuplicates                                
         .alias("v")

My guess is probably the hash partitioning

Any idea how to fix this?

---
Graphframe version: 
`"graphframes" % "graphframes" % "0.2.0-spark2.0-s_2.11",`

**Comment by felixcheung** [comment]
To confirm, you are running with Spark 2.0.0?
What is the data source for `genes` in your example? Do you see this issue with Spark without GF?

**Comment by mavencode01** [comment]
@felixcheung we running Spark 2.0.2 and data source is from aws RDS.

We don't have such issue without the GF, but I think is easily reproducible if you load large number of data...the bugs shows up.

**Comment by felixcheung** [comment]
have you tried GF 0.3 release? there's a new connected component implementation

**Comment by mavencode01** [comment]
Yeah, tried but gave up because of the checkpoint requirements in 3.0. 

I'm running on a standalone clusters and checkpoint fails because it cannot access the checkpoint directory from another node.

Will try to get an EMR cluster to setup and try.

**Comment by jkbradley** [comment]
@mavencode01  When you write "143M records," is that the number of vertices or edges?  What are the numbers of vertices & edges?

Also, do you have very high degree vertices (10-100M)?  If so, then the new implementation should be much more scalable.  Any luck with a cluster which allows attaching an external filesystem for checkpointing?

**Comment by SemyonSinchenko** [comment]
Closed due to inactivity
